pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -325,10 +326,16 @@ func changefeedPlanHook(
 		}
 	}
 
+	var e *kvpb.BatchTimestampBeforeGCError
 	rowFnLogErrors := func(ctx context.Context, resultsCh chan<- tree.Datums) error {
 		err := rowFn(ctx, resultsCh)
 		if err != nil {
 			logChangefeedFailedTelemetryDuringStartup(ctx, description, failureTypeForStartupError(err))
+			if errors.As(err, &e) && opts.HasStartCursor() {
+				err = errors.Newf("could not create changefeed: cursor %s is older than the GC threshold %d", opts.GetCursor(), e.Threshold.WallTime)
```

Comment:
My preference would be to wrap the original error (with `errors.Wrapf`) instead of replacing it